app/controllers/assets_controller.rb

```diff
@@ -190,11 +191,25 @@ def create_wopi_file
     render_403 && return unless %w(docx xlsx pptx).include?(params[:file_type])
 
     # Asset validation
+    original_filename = "#{params[:file_name]}.#{params[:file_type]}"
     file = Paperclip.io_adapters.for(StringIO.new)
-    file.original_filename = "#{params[:file_name]}.#{params[:file_type]}"
+    file.original_filename = original_filename
     file.content_type = wopi_content_type(params[:file_type])
     asset = Asset.new(file: file, created_by: current_user, file_present: true)
 
+    # Filename length validation (this cannot be checked by Paperclip,
+    # as it depends on OS)
+    if original_filename.length > Constants::FILENAME_MAX_LENGTH
+      render json: {
+        message: {
+          file: I18n.t(
+            'assets.create_wopi_file.errors.file_name_too_long',
+            limit: Constants::FILENAME_MAX_LENGTH
+          )
+        }
+      }, status: 400 and return
+    end
```

Comment:
I believe this could easily be put into `asset.rb` into `:wopi_file_creation` context. See the line 213, it already checks for presence of filename, you'd just have to perform the same check in `name_should_not_be_empty_without_extension` and would be less mouthful and cleaner code, since only the check on 213 would be performed. (probably the method name should be changed then to reflect the changes).